astropy/cosmology/io/model.py

```diff
@@ -238,7 +238,7 @@ def to_model(cosmology, *_, method):
     # make Model class
     CosmoModel = type(clsname, (_CosmologyModel,), {**attrs, **params})
     # override __signature__ and format the doc.
-    setattr(CosmoModel.evaluate, "__signature__", msig)
+    CosmoModel.evaluate.__signature__ = msig
```

Comment:
I've run into issues with making this change over in ASDF. You should check that your comfortable wit this change.

Reply:
Can you elaborate? I thought ``setattr(instance, name, value)`` is the functional form of ``instance.__setattr__(name, value)``. When would this break code?

Reply:
I was surprised too; however, I did not investigate the issue too closely in ASDF. You can find the two issues I had in ASDF:

1. https://github.com/asdf-format/asdf/blob/ae9b77cdc8d69f1f1fd136ff48c9a523a86c18af/asdf/types.py#L85
2. https://github.com/asdf-format/asdf/blob/ae9b77cdc8d69f1f1fd136ff48c9a523a86c18af/asdf/types.py#L149

Reply:
Oh. I think this how Python handles "__X" access. If you look how those objects are stored, Python adds class information to the attribute name!

Reply:
Personally these are rare enough I'd rather have the rule. If we ever run into this in astropy and don't want to add a noqa we can open an issue for ruff / flake8-bugbear